lib/puppet/network/formats.rb

```diff
@@ -23,13 +23,26 @@ def render_multiple(instances)
 end
 
 Puppet::Network::FormatHandler.create_serialized_formats(:yaml) do
+  # Cannot be a constant since this handler is initialized before some of the classes exists
+  def allowed_yaml_classes
+    @allowed_yaml_classes ||= [
+      Puppet::Node::Facts,
+      Puppet::Node,
+      Puppet::Transaction::Report,
+      Puppet::Resource,
+      Puppet::Resource::Catalog,
+      Symbol,
+      Time,
+    ].freeze
```

Comment:
We need to allow any class that puppet may have written as YAML previously, e.g. when upgrading from an older version. Are you sure this is the complete list of classes that puppet may have serialized as YAML? For example, I see other classes including `PsychSupport` that aren't in the above list.

I assume we don't need to include things like `Puppet::Transaction::Event` which are contained within a catalog?

Reply:
The remaining classes seems to be `Puppet::Resource::Status`, `Puppet::Relationship`, `Puppet::Util::Metric`, `Puppet::Indirector::Request`, `Puppet::Util::SimpleGraph`, and `Puppet::Util::Log`. I might have missed something but I couldn't find any places where they were read with yaml with the expectation deserialize into an instance. If you have concerns about any of them, then we can add them to the list. But I think the shorter the list, the easier it is to control the safety. Ideally, the list should be empty.